packages/connectivity/src/scp-cf/destination/destination-from-register.ts

```diff
@@ -0,0 +1,140 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { Cache, IsolationStrategy } from '../cache';
+import { decodeJwt } from '../jwt';
+import { userId } from '../user';
+import { getXsuaaServiceCredentials } from '../environment-accessor';
+import { parseSubdomain } from '../subdomain-replacer';
+import { Destination, DestinationAuthToken } from './destination-service-types';
+import { DestinationFetchOptions } from './destination-accessor-types';
+import { DestinationCache } from './destination-cache';
+import {
+  addProxyConfigurationInternet,
+  ProxyStrategy,
+  proxyStrategy
+} from './proxy-util';
+
+const logger = createLogger({
+  package: 'connectivity',
+  messageContext: 'register-destination'
+});
+/**
+ * @internal
+ */
+export const registerDestinationCache = DestinationCache(
+  new Cache<Destination>(undefined)
+);
+
+type RegisterDestinationOptions = Pick<
+  DestinationFetchOptions,
+  'jwt' | 'isolationStrategy'
+>;
+
+/**
+ * Set a given destination in the `destinations` environment variable.
+ *
+ * Throws an error if a destination with the same name as the given test destination already exists.
+ * @param destination - A destination to add to the `destinations` cache
+ * @param options - Options how to cache the destination
+ */
+export function registerDestination(
+  destination: DestinationWithName,
+  options?: RegisterDestinationOptions
+): void {
+  if (!destination.name || !destination.url) {
+    throw Error(
+      'The registerDestination function requires a destination name and url.'
```

Comment:
I would make this less technical -
```suggestion
      'Registering destinations requires a destination name and url.'
```